adapters/repos/db/vector/hnsw/deserializer.go

```diff
@@ -564,24 +565,29 @@ func (d *Deserializer) ReadPQ(r io.Reader, res *DeserializationResult) error {
 		UseBitsEncoding:     useBitsEncoding != 0,
 	}
 	var encoderReader func(io.Reader, *DeserializationResult, uint16) (compressionhelpers.PQEncoder, error)
+	var totalRead int
 	switch encoder {
 	case compressionhelpers.UseTileEncoder:
 		encoderReader = d.ReadTileEncoder
+		totalRead = 51 * int(res.PQData.M)
```

Comment:
it will always be 51? right?

Reply:
51 is the amount of bytes each tile encoder writes. Total read will be 51 times the amount of encoders (res.PQData.M). For the case of the Means encoder it is different. It depends on the centroids per dimension.

Reply:
Thank you for the explanation!